vj4/handler/domain.py

```diff
@@ -183,14 +194,17 @@ class DomainRoleHandler(base.OperationHandler):
                                              fields={'uid': 1, 'role': 1}):
       if 'role' in dudoc:
         rucounts[dudoc['role']] += 1
+    # built-in roles are displayed additionally so that we don't need to use get_all_roles() here
     roles = sorted(list(self.domain['roles'].keys()))
     self.render('domain_manage_role.html', rucounts=rucounts, roles=roles)
 
   @base.require_perm(builtin.PERM_EDIT_PERM)
   @base.require_csrf_token
   @base.sanitize
-  async def post_set(self, *, role: str, perm: int=builtin.DEFAULT_PERMISSIONS):
-    await domain.set_role(self.domain_id, role, perm)
+  async def post_set(self, *, role: str):
```

Comment:
Because we does not support adding a role from copying a specific role. Even if it is supported one day, the parameter should not be `perm` as well, it should be something like `existing_role`, or a perm array. Anyway, the `perm` parameter is not going to be used in the foreseeable future